backend/src/services/contact/contact.service.ts

```diff
@@ -28,19 +29,53 @@ export class ContactService {
     private readonly searchService: SearchService,
   ) {}
 
+  async caculateContactScore(contacts: Contact[]): Promise<AbstractContact[]> {
+    const contactsWithScore = contacts.map((contact) => {
+      if (contact.notes.length === 0) {
+        return { ...contact, totalScore: 0 };
+      }
+
+      const lastStatus =
+        contact.statuses[contact.statuses.length - 1].status.status;
+
+      const notesUsedForScoreCalculation = contact.notes.filter(
+        (note) => note.score !== null && note.status.status === lastStatus,
+      );
+      const scores = notesUsedForScoreCalculation.map((note) => {
+        return note.score;
+      });
+
+      const sumScores = scores.reduce((acc, s) => {
+        return acc + s;
+      }, 0);
```

Comment:
For this part, you don't need to map the `notesUsedForScoreCalculation` into another const and reduce the const.
You can directly use the `reduce` on the `notesUsedForScoreCalculation`.